src/Middleware/SocialAuthMiddleware.php

```diff
@@ -225,7 +225,8 @@ protected function _handleCallbackAction(ServerRequest $request): Response
 
         $user->unset($config['fields']['password']);
 
-        $event = $this->dispatchEvent(self::EVENT_AFTER_IDENTIFY, ['user' => $user]);
+        $data = ['user' => $user, 'session' => $request->getSession()];
```

Comment:
I am not happy with passing session instance as event data. As I commented on the other issue I would rather use query string.

> I wonder if that could also conflict with other query strings and stuff.

The AuthComponent and new Auth plugin uses `redirect` query string var to retrain the URL to redirect to after successful login (change done by yourself). No one has complained about that causing any issues, so I don't see this being any different.

> Or we would need to make the query string a bit more unique and less collision likely.

Yes, I am open to suggestions for what var name to use.

Reply:
redirect is a global default expectation IMO.
that one not so much (yet)

How about both possible?
I would rather use the internal session handling than the query string for that case, and it is as "safe" as the other operations using direct session access IMO, e.g. getUser() event.
We can however document primarily the query string one if we want to.

Reply:
Also redirect is a different topic altogether (I am well aware of that being my thing^^).
It is added to the login action only and also removed again when redirecting to the actual controller actions across the whole app and plugins.

The query string would directly be visible and can have annoying side effects
E.g. reload would give the same message again if you don't redirect and remove that one.

So I really would advise against using query string here.

Reply:
Okay, but I have something a bit different in mind to achieve the same. I will make a PR soon.